libtracepoint-control-cpp/src/TracepointSession.cpp

```diff
@@ -1021,221 +1065,38 @@ TracepointSession::SavePerfDataFile(
 }
 
 _Success_(return == 0) int
-TracepointSession::DisableTracePointImpl(PerfEventMetadata const& metadata) noexcept
+TracepointSession::DisableTracepointImpl(PerfEventMetadata const& metadata) noexcept
 {
     int error;
 
-    auto const it = m_tracepointInfoByCommonType.find(metadata.Id());
-    if (it == m_tracepointInfoByCommonType.end())
+    auto const existingIt = m_tracepointInfoByCommonType.find(metadata.Id());
+    if (existingIt == m_tracepointInfoByCommonType.end())
     {
         error = ENOENT;
     }
-    else if (it->second.m_enableState == TracepointEnableState::Disabled)
-    {
-        error = 0;
-    }
     else
     {
-        error = IoctlForEachFile(it->second.m_bufferFiles.get(), m_bufferCount, PERF_EVENT_IOC_DISABLE, nullptr);
-        it->second.m_enableState = error
-            ? TracepointEnableState::Unknown
-            : TracepointEnableState::Disabled;
+        error = SetTracepointEnableState(existingIt->second, false);
```

Comment:
Note: The IoctlForEachFile was the same for the Disable and Enable cases, so moved into a helper method.